torch/_ops.py

```diff
@@ -770,11 +770,10 @@ def handler(*args, **kwargs):
                     # TODO: This path is slow, should generally encourage this
                     # case to not happen
                     return self._op_dk(key, *args, **kwargs)
-                # TODO(voz): The idea behind this is that we do not yet support dispatch by key + mode, only key.
-                return self.python_key_mode_table[curr_mode](*args, **kwargs)
 
-            self._dispatch_cache[key] = handler
```

Comment:
When inputs are normal script object, it goes through C++ dispatcher then go back to python in `_get_dispatch`.  Previously, this logic of handling mode is never trigged because python_key_mode_table is empty ([L755](https://github.com/pytorch/pytorch/pull/123367/files#diff-9453b6b52a54783beec3dd1c60248620f61c3a524d404a188af17bbdf6be3d9eL755)) if the op is not higher order ops (higher order op dispatch completely in python so they won't go through `_get_dispatch`). 

Even though there is not implementation for these keys for custom op, it seems that there're backend fallback implementations for these keys so they can be traced fine previously. 

For TorchBindOpOverload, we'll by default implement these dispatch mode keys so the python_key_mode_table is not empty any more. when inputs are not fakified, functionalizing/tracing through them can trigger the `_get_dispatch` logic here. We therefore need to add proper fake mode support in this case.

Reply:
I'm confused, are you saying we need to invalidate the cache?